app/common/state/aboutNewTabState.js

```diff
@@ -29,6 +29,21 @@ const sortCountDescending = (left, right) => {
   if (left.get('count') > right.get('count')) return -1
   return 0
 }
+const removeDuplicateDomains = (list) => {
+  const siteDomains = []
+  return list.filter((site) => {
+    try {
+      const hostname = require('url').parse(site.get('location')).hostname
+      if (!siteDomains.includes(hostname)) {
```

Comment:
Pls use a Set here, or just indexing inside an object is fine. You can make this O(N) or O(NlogN) depending on complexity of lookups in an ES6 Set.  Which is much better than the current O(N^2).